Libplanet/HashDigest.cs

```diff
@@ -83,9 +83,9 @@ public ImmutableArray<byte> ByteArray
         {
             get
             {
-                if (_byteArray.IsDefault)
+                if (_byteArray == default)
                 {
-                    _byteArray = new byte[Size].ToImmutableArray();
+                    return (new byte[Size]).ToImmutableArray();
```

Comment:
How about cache and re-use new byte[Size] as static?

Reply:
It's a good idea. Implementing it!